### PR TITLE
Feature/ch3790/cp1 phjob controller basic workload

### PR DESCRIPTION
**Pull request by popcornylu**
## Links

- https://app.clubhouse.io/infuseai/story/3790/cp1-phjob-controller-basic-workload
- https://app.clubhouse.io/infuseai/story/3591/cp1-provide-sample-phjob-definition

## What does this MR do?

This first phjob controller PR. The basic journey is

1. user create a phjob
2. controller create a pod according to phjob spec and some logic according to the result of graphql
3. On pod status changed, feedback the condition to phjob
4. It would also mount project volume if the specified group is shared volume enabled.

### CODE REVIEW

**Comment by jackklpan** [comment]
Please update config.yaml.example

**Comment by popcornylu** [comment]
The PR also include
https://app.clubhouse.io/infuseai/story/3591/cp1-provide-sample-phjob-definition

**Comment by jackklpan** [comment]
* It cannot access `global` instancetype and image
* When instance didn't set `Memory Request`, graphql return `""` and controller will crash

File: controllers/phjob_controller.go [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+				phJob.Status.Reason = err.Error()
+				log.Error(err, "failed to create pod")
+			}
+
+			if err = r.Status().Update(ctx, phJob); err != nil {
+				log.Error(err, "failed to update PhJob status")
+				return ctrl.Result{}, err
+			}
+		} else {
+			log.Info("Pod not found but not necessary ot create")
```
Comment (jackklpan) [inline controllers/phjob_controller.go]:
```suggestion
			log.Info("Pod not found but not necessary to create")
```

File: config.yaml.example [hunk 2]
```diff
@@ -12,3 +12,7 @@ customImage:
       limits:
         cpu: 1000m
         memory: 500Mi
+
+jobSubmission:
+  graphqlEndpoint: https://hub.celu.dev.primehub.io/graphql
+  graphqlSecret: b3f0c00afb6b8e37e3f9ea05c1695ece6f8758411950ed711a8f88687fb22c84
```
Comment (jackklpan) [inline config.yaml.example]: Do we need to change to sth more general rather than real one?

File: pkg/graphql/spawn.go [hunk 3]
```diff
@@ -0,0 +1,191 @@
+package graphql
+
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+import (
```
Comment (jackklpan) [inline pkg/graphql/spawn.go]: Maybe can remove this redundant import.

Comment (popcornylu) [reply pkg/graphql/spawn.go]: Done

File: config.yaml.example (hunk 2 above)
Comment (popcornylu) [reply config.yaml.example]: Done.

File: controllers/phjob_controller.go (hunk 1 above)
Comment (popcornylu) [reply controllers/phjob_controller.go]: Fixed

**Comment by popcornylu** [comment]
@jackklpan All above issues has been fixed. Please check again. Thanks for your valuable change request.

**Pull request merged**